packages/@lwc/engine-core/src/libs/signal-tracker/index.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { isFalse, isUndefined } from '@lwc/shared';
+import { Signal } from '@lwc/signals';
+import { logWarnOnce } from '../../shared/logger';
+
+/**
+ * This map keeps track of objects to signals. There is an assumption that the signal is strongly referenced
+ * on the object which allows the SignalTracker to be garbage collected along with the object.
+ */
+const TargetToSignalTrackerMap: WeakMap<Object, SignalTracker> = new WeakMap();
```

Comment:
```suggestion
const TargetToSignalTrackerMap = new WeakMap<object, SignalTracker>();
```
1. you can provide generics directly to the constructor
2. `Object` -> `object`, as described below

Reply:
Nice, TIL!